pkg/asset/installconfig/aws/ec2.go

```diff
@@ -5,11 +5,33 @@ import (
 	"fmt"
 	"time"
 
+	awsv2 "github.com/aws/aws-sdk-go-v2/aws"
+	ec2v2 "github.com/aws/aws-sdk-go-v2/service/ec2"
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/ec2"
 )
 
+func getEC2Client(cfg awsv2.Config) *ec2v2.Client {
+	return ec2v2.NewFromConfig(cfg)
+}
+
+// GetRegions get all regions that are accessible.
+func GetRegions(ctx context.Context, config awsv2.Config, allRegions bool) ([]string, error) {
+	client := getEC2Client(config)
```

Comment:
We do need to set custom ec2 endpoints for this if it exists, so we do need to figure how we're generally going to handle the clients. With the V2 sdk the endpoint is always set on the client, so I think it would be a good idea to have a centralized way of retrieving or instantiating the client, so that we are less likely to forget setting custom endpoints whenever we use the client. 

One idea is to save the clients into the metadata and then pass the client into functions like this.